Rubberduck.Main/VersionCheck.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Rubberduck.VersionCheck
+{
+    public class VersionCheck : IVersionCheck
+    {
+        private readonly Lazy<Version> _currentVersion;
+        public VersionCheck()
```

Comment:
Maybe it would be better to leave the class in `Core` and constructor inject the assembly or a function providing the assembly or version. The `Main` assembly can be retrieved conveniently inside the IoC installer since it is located in that assembly.